sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp

```diff
@@ -26,7 +26,12 @@ namespace Azure { namespace Core { namespace Http {
   struct CurlTransportSSLOptions
   {
     bool AllowBeast = false;
-    bool NoRevoke = false;
+    /**
+     * @brief The libcurl default behavior is false for disabling the revocation list check.
+     * However, this is overriden for the Azure SDK to disable the revocation list check by default.
```

Comment:
`false for disabling...` - I am already lost :p (also libcurl doesn't use a boolean here)

```suggestion
     * @brief By default this is set to true which disables the revocation list check by overriding the out-of-the-box libcurl behavior. Setting this to false will enable revocation list checks.
```

Reply:
The default is what libcurl does by default, not what the Azure SDK does.  That's why we need to mention what's the default and what the Azure SDK is doing.

Reply:
If the caller doesn't set an SSL option and don't pass in a curl transport option, what behavior will they get? That default behavior is what we need to document since that is the relevant part for our users.

Reply:
In this case, I don't agree.

When it comes to the SDK options, that is fine and correct.

But when it comes to the options that the SDK will turn ON/OFF from a third party lib like libcurl, I strongly believe that the SDK should honor whatever defaults are for the 3rd party lib. And in the case of overriding that value to a different default value, this should be told as part of the docs (I would even expect a nice reason in the docs about why the SDK decided to change the defaults from the 3rd party).

To me, it is the same as what we do for the Service's rest API that takes optional values. The SDK will never choose what those default values are. The SDK always honors the Service's defaults by never overriding.

To me, the SDK is just helping you to group all the tools and data that you need to interact with Azure without taking deliberate decisions for the user. The only defaults or behavior set by the SDK is for features owned by the SDK alone, like the retry, or things like that.

Reply:
The analogy between the third party library and the service rest APIs doesn't hold up because talking to the service is the primary (and arguably only) purpose of the SDK and that is what customers care about. The settings and use of libcurl within the SDK itself to achieve that goal is largely an implementation detail that most people won't care about.

No one will come to the Azure SDK for C++ to wonder what libcurl settings can I play with today or figure out its particular nuances. They will come to it to talk to the storage service and upload and download blob data. That is the customer scenario and motivation. Trying to minimize the underlying complexity and machinery within the SDK (though top-of-mind and critical for us as its authors) needs to largely be hidden from the customer and only shown as part of progressive disclosure.

Reply:
What you mention applies for some end-users profiles, but not for all profiles.

We can't mix the profiles!  Think about the end-user profile that will use and care about this setting/featue for libcurl. 

Consider we are adding this feature based on end-user request. Then, you can't say this end-user don't care about it, if he is asking for it.

This end-user is asking specifically for this feature to be configurable, he knows in advance how this feature works and how to set it up with libcurl alone. Note that the end-user is not asking for it as a general feature for the SDK because he did it before with another language SDK or something like that. The end-user is specifically asking for the libcurl option that he needs. He is not interested in how to do the same for another HTTP stack or another language.
That's why, if we are changing the way this option work, we need to disclosure that to the end-user.

We are building this feature for a specific end-user profile.  In this case, why do we want to build the feature for the general end-user usage/understanding (which we already considered that doesn't really care about it).

Reply:
> This end-user is asking specifically for this feature to be configurable, he knows in advance how this feature works and how to set it up with libcurl alone. Note that the end-user is not asking for it as a general feature for the SDK because he did it before with another language SDK or something like that. The end-user is specifically asking for the libcurl option that he needs. He is not interested in how to do the same for another HTTP stack or another language.
That's why, if we are changing the way this option work, we need to disclosure that to the end-user.

Is the framing of this request a theoretical end-user ask that could come up or one we have actually seen presented this way? For this particular feature, the request would largely be motivated by:
- observed performance slow downs in user applications or denial of service
- default use by looking at code samples not being good enough

https://twitter.com/lapcatsoftware/status/1326990296412991489

The end-user looking specifically about libcurl capabilities and interested in the technical reasons/map of options is likely to be in an extreme minority. Most people will arrive at feature requests from some observed application level pain-point or gap. They probably wouldn't even notice or realize that the SDK is using libcurl under the covers and how we have configured it to be the root cause of the issue.

We fixed the issue by turning CRL checks off by default and now most people wouldn't worry about it, care to turn it on or even need to know about it, and probably wouldn't benefit from this option being available to begin with. It is there for folks who understand and want that extra security, understand the trade-offs, and can't use OCSP or other mechanism.

https://docs.microfocus.com/NNMi/10.30/Content/Administer/NNMi_Deployment/Advanced_Configurations/Cert_Validation_CRL_and_OCSP.htm#:~:text=Certificate%20Revocation%20List%20(CRL)%20%2D,service%20called%20an%20OCSP%20responder.

Reply:
The end-user is here:
https://github.com/Azure/azure-storage-cpp/issues/295

Since we didn't see this issue with libcurl on Windows, it means that winSSL has this disabled by default.

Reply:
> The end-user is here:
Azure/azure-storage-cpp#295

Precisely. This is how customers observe and react to the issue, not from a perspective of what settings of libcurl or winhttp are being turned on/off.

> I started working with azure-storage-cpp and everything was fine until I started testing my project on Azure VM.
My problem is that any call to the Azure Storage server takes an unacceptable amount of time to execute. Each server request takes ~100 seconds.